squad/frontend/static/squad/controllers/charts.js

```diff
@@ -719,6 +720,13 @@ function ChartsController($scope, $http, $location, $compile, ChartPanel, DATA)
                 if (! metric.dynamic) {
                     metric.drawn = true
                 }
+                var canvas_id = document.getElementById(metric.name).id
+                var button_id = canvas_id + "-anchor"
+                var canvas_div = document.getElementById(metric.name+'-div').children[0]
+                var download_elem = "<a id='"+ button_id + "' ng-click='downloadChartImage(\""+ canvas_id + "\""+
```

Comment:
@mwasilew I am not sure we can have a permalink pre-created since the href is populated once the 'download' button is clicked first time...hmm..will look into this again later tomorrow.

Reply:
@amrohassaan then we need an API to do that. The idea is to allow squad client downloading images for the purpose of reporting.